dial_queue.go

```diff
@@ -0,0 +1,269 @@
+package dht
+
+import (
+	"context"
+	"errors"
+	"math"
+	"sync"
+	"time"
+
+	peer "github.com/libp2p/go-libp2p-peer"
+	queue "github.com/libp2p/go-libp2p-peerstore/queue"
+)
+
+var DialQueueMinParallelism = 6
+var DialQueueMaxParallelism = 20
+var DialQueueMaxIdle = 5 * time.Second
+var DialQueueScalingMutePeriod = 1 * time.Second
```

Comment:
This pattern in the PL codebases is irksome. I think generally global constants should only be vars if there's no alternative. Also consider `var ( ... )` or `const ( ... )`.

Reply:
we usually do put it in a enclosing var, as it should be done here too.